src/components/z-carousel/index.tsx

```diff
@@ -1,36 +1,284 @@
+          return;
+        }
+
+        this.current = entryIndex;
+        this.scrollingTo = null;
+      },
+      {
+        root: this.itemsContainer,
+        threshold: 0.5,
+      }
+    );
+
+    this.items.forEach((element) => this.intersectionObserver.observe(element));
+  }
+
+  private onPrev(): void {
+    if (this.single) {
+      this.current = Math.max(0, this.current - 1);
+      return;
+    }
+
+    const scroller = this.itemsContainer;
+    if (!scroller) {
+      return;
+    }
+
+    scroller.scrollBy({
+      left: -scroller.clientWidth / 2,
+      behavior: "smooth",
+    });
+  }
+
+  private onNext(): void {
+    if (this.single) {
+      this.current = Math.min(this.current + 1, this.items.length - 1);
+
+      return;
+    }
+
+    const scroller = this.itemsContainer;
+    if (!scroller) {
+      return;
+    }
+
+    scroller.scrollBy({
+      left: scroller.clientWidth / 2,
+      behavior: "smooth",
+    });
+  }
+
+  /**
+   * Check if footer can be rendered.
+   * @returns {boolean}
+   */
+  private canShowFooter(): boolean {
+    return (
+      this.arrows === CarouselArrowsPosition.BOTTOM ||
+      this.progress === CarouselProgressMode.DOTS ||
+      this.progress === CarouselProgressMode.NUMBERS
+    );
+  }
+
+  /**
+   * Set current item to passed index.
+   * @param {number} index Index to set
+   * @returns {void}
+   */
+  private goTo(index): void {
+    if (this.current === index) {
+      return;
+    }
+
+    this.current = index;
+  }
+
+  render(): HTMLDivElement | HostElement {
     if (this.isLoading) {
-      return (
+      return <Host>
+        {this.label && <div class="heading-4 z-carousel-title">{this.label}</div>}
         <div style={{ height: `${this.ghostLoadingHeight}px` }}>
           <z-ghost-loading></z-ghost-loading>
-          <div style={{ display: "none" }}>
+          <div class="loading-items-container">
             <slot />
           </div>
         </div>
-      );
+      </Host>;
     }
+
     return (
-      <ul class="z-carousel-items-container">
-        <slot />
-      </ul>
+      <Host>
+        <div class="z-carousel-container">
+          {this.label && <div class="heading-4 z-carousel-title">{this.label}</div>}
+          {this.arrows === CarouselArrowsPosition.OVER && (
+            <z-button
+              data-direction="prev"
+              icon="chevron-left"
+              onClick={this.onPrev.bind(this)}
+            />
+          )}
+          <ul class="z-carousel-items-container">
+            <slot />
+          </ul>
+          {this.arrows === CarouselArrowsPosition.OVER && (
+            <z-button
+              data-direction="next"
+              icon="chevron-right"
+              onClick={this.onNext.bind(this)}
+            />
+          )}
+        </div>
+
+        {this.canShowFooter() && (
+          <div class="z-carousel-footer">
+            {this.arrows === CarouselArrowsPosition.BOTTOM && (
```

Comment:
Vale la stessa considerazione fatta sopra.

Reply:
mmh non so secondo me creare un paio di variabili non migliora la leggibilità del codice, e valutare due variabili così è un "guadagno" in performance irrisorio. Visto che non si tratta di fare chiamate di rete o evitare computazioni onerose, garantirei una leggibilità maggiore.